src/gles_fns.rs

```diff
@@ -0,0 +1,1311 @@
+// Copyright 2014 The Servo Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+pub struct GlesFns {
+    ffi_gl_: GlesFfi,
+}
+
+impl GlesFns
+{
+    fn new(ffi_gl_: GlesFfi) -> Arc<GlesFns> {
```

Comment:
Today, I tried to remove Arc usage, but I failed to do it. Compiler still emitted errors like the following. And I am still not sure how to deliver ref to background thread.

------------------------
 0:03.24 error[E0502]: cannot borrow `self.device` as mutable because it is also borrowed as immutable
 0:03.24    --> /home/sotaro/quantum_gfx_src_6/graphics/gfx/webrender/src/renderer.rs:947:25
 0:03.24     |
 0:03.24 900 |         let _gm = GpuMarker::new(self.device.get_gl() ,"texture cache update");
 0:03.24     |                                  ----------- immutable borrow occurs here
 0:03.24 ...
 0:03.24 947 |                         self.device.deinit_texture(texture_id);
 0:03.24     |                         ^^^^^^^^^^^ mutable borrow occurs here
 0:03.24 ...
 0:03.24 952 |     }
 0:03.24     |     - immutable borrow ends here